### PR TITLE
bump alertmanager to 0.22.2

### PR DESCRIPTION
**Pull request by TheoBrigitte**
We are now using alertmanager 0.22.2, so PMO needs to use the same version internally to avoid any problem.

This PR was created by following the procedure written here https://github.com/giantswarm/prometheus-meta-operator/blob/master/README.md#how-to-update-upstream-code

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/giantswarm/micrologger v0.5.0
 	github.com/giantswarm/operatorkit/v4 v4.3.1
 	github.com/giantswarm/versionbundle v0.2.0
-	github.com/go-kit/log v0.1.0
+	github.com/go-kit/kit v0.10.0
```
Comment (mig4) [inline go.mod:17]: So this is needed after all?

Comment (TheoBrigitte) [reply go.mod:17]: that's required by alertmanager/config: https://github.com/prometheus/alertmanager/blob/44f8adc06af5101ad64bd8b9c8b18273f2922051/config/coordinator.go#L21

Comment (mig4) [reply go.mod:17]: Right, got it.

**Comment by TheoBrigitte** [comment]
This was tested successfully on ginger.

**Pull request merged**